python/pyarrow/tests/test_compute.py

```diff
@@ -2085,6 +2085,18 @@ def _check_temporal_rounding(ts, values, unit):
         expected = ts.dt.round(frequency)
         np.testing.assert_array_equal(result, expected)
 
+    if ta.type.tz is None and unit != "day":
```

Comment:
What happens if `ta.type.tz` is not None, or if `unit` is days?

Reply:
I've made a mistake of overwriting the `frequency` parameter whenever unit was `day`. Now his is only disabled for zoned times, which I think is as far we can go with such naive logic.